format/spec.md

```diff
@@ -241,7 +241,9 @@ Struct evolution requires the following rules for default values:
 
 #### Column Projection
 
-Columns in Iceberg data files are selected by field id. The table schema's column names and order may change after a data file is written, and projection must be done using field ids. If a field id is missing from a data file, its value for each row should be `null`.
+Columns in Iceberg data files are selected by field id. The table schema's column names and order may change after a data file is written, and projection must be done using field ids.
+
+When a projected column has an [identity partition transform](#partition-transforms) applied to it for a data file, the value from the [manifest file](#manifests) must be used for that column (i.e. the column should not be read from the data file). This is to support tables that were migrated from other table formats (notably Hive) that do not write partition values to data files. Otherwise, if a field id is missing from a data file, its value for each row should be `null`.
```

Comment:
Question, if projected column has other partition transform, then also the value must be read from manifest file itself right? I am not clear on why this is specific to identity partition transform?

Reply:
So, we can update "identity partition transform" ->  "partition transform" and remove mention about hive and consider default values. 

How about like this?

> When a projected column has [partition transform] (#partition-transforms) applied to it for a data file, the value from the [manifest file](#manifests) must be used for that column (i.e. the column should not be read from the data file). Otherwise, if a field id is missing from a data file, its value for each row should be `null` unless a default value is specified.

Reply:
changed wording as suggested by @RussellSpitzer